src/joining.cu

```diff
@@ -139,41 +143,31 @@ DEF_OUTER_JOIN(f64, int64_t)
   if (T1 == GDF_DATE64)    { JOIN_HASH_T2(int64_t, l1, r1, T2, l2, r2, T3, l3, r3) } \
   if (T1 == GDF_TIMESTAMP) { JOIN_HASH_T2(int64_t, l1, r1, T2, l2, r2, T3, l3, r3) }
 
-// multi-column join function
-template <JoinType join_type>
-gdf_error multi_column_join(int num_cols, gdf_column **leftcol, gdf_column **rightcol, gdf_join_result_type **out_result)
+/* --------------------------------------------------------------------------*/
+/** 
+ * @Synopsis Computes the Join result between two tables using the hash-based implementation. 
+ * 
+ * @Param num_cols The number of columns to join
+ * @Param leftcol The left set of columns to join
+ * @Param rightcol The right set of columns to join
+ * @Param out_result The result of the join operation. The first n/2 elements of the
+   output are the left indices, the last n/2 elements of the output are the right indices.
+   @tparam join_type The type of join to be performed
+ * 
+ * @Returns Upon successful computation, returns GDF_SUCCESS. Otherwise returns appropriate error code 
+ */
+/* ----------------------------------------------------------------------------*/
+template <JoinType join_type, 
+          typename size_type>
+gdf_error hash_join(size_type num_cols, gdf_column **leftcol, gdf_column **rightcol, gdf_join_result_type **out_result)
```

Comment:
Have you talked with Kumar Antish who is working on chaning it so that this output is actually a pair of gdf_columns as opposed to gdf_join_result_type?